kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -255,15 +262,16 @@ public boolean handle(Request request, Response response, Callback callback) thr
         };
     }
 
-    private SslContextFactory.Server getSSLContextFactory() {
+    private SslContextFactory.Server getSSLContextFactory() throws GeneralSecurityException, IOException {
         SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+        sslContextFactory.setTrustStore(KafkaAgentUtils.jksTrustStore(caCertSecret));
 
-        sslContextFactory.setKeyStorePath(sslKeyStorePath);
-        sslContextFactory.setKeyStorePassword(sslKeyStorePassword);
-        sslContextFactory.setKeyManagerPassword(sslKeyStorePassword);
+        byte[] random = new byte[24];
+        RANDOM.nextBytes(random);
+        String password = Base64.getUrlEncoder().withoutPadding().encodeToString(random).substring(0, 32);
```

Comment:
can you elaborate the rationale behind the 24 bytes choice and the substring up to 32?

Reply:
I guess it should be fine, thanks.